ReaderApp.js

```diff
@@ -1442,13 +1479,14 @@ class ReaderApp extends React.PureComponent {
       }
       const nextFilter = new LinkFilter(name, heName, collectiveTitle, heCollectiveTitle, nextRefList, nextHeRefList, category);
 
-      this.state.linkRecentFilters[filterIndex] = nextFilter;
+      const newLinkRecentFilters = [...this.state.linkRecentFilters];
```

Comment:
wrap this in setState callback and use prevState when calculating `newLinkRecentFilters`